.github/workflows/docker.yml

```diff
@@ -10,21 +10,67 @@ on:
 
 jobs:
   build:
+    permissions:
+      id-token: write
     strategy:
       matrix:
         runner: ["ubuntu-latest", "buildjet-2vcpu-ubuntu-2204-arm"]
     runs-on: ${{ matrix.runner }}
     steps:
       - name: checkout
         uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # tag=v4.1.1
+      - name: Login to Stackable Harbor
+        uses: docker/login-action@343f7c4344506bcbf9b4de18042ae17996df046d # tag=v3
+        with:
+          registry: oci.stackable.tech
+          username: robot$sdp+github-action-build
+          password: ${{ secrets.HARBOR_ROBOT_SDP_GITHUB_ACTION_BUILD_SECRET }}
+      - name: Set up Cosign
+        uses: sigstore/cosign-installer@9614fae9e5c5eddabb09f90a270fcb487c9f7149 # tag=v3.3.0
+      - name: Set up syft
+        uses: anchore/sbom-action/download-syft@24b0d5238516480139aa8bc6f92eeb7b54a9eb0a # tag=v0.15.5
       - name: buildstep
         id: buildstep
         shell: bash
         env:
           DOCKER_USER: github
           DOCKER_PASSWORD: ${{ secrets.NEXUS_PASSWORD }}
-        run: make push-ubi8-rust-builder
+        run: |
+          make push-ubi8-rust-builder
+
+          # TODO: obtain architecture from GH directly rather than from arch
```

Comment:
What about this TODO?

Reply:
It's a technical detail want to leave it for later, if this is fine for you. Doesn't have impact right now.

Reply:
https://cli.github.com/manual/gh_help_environment

thought it would have an env var but I was wrong. Removing the todo

Reply:
Done!

Reply:
See here: https://docs.github.com/en/actions/learn-github-actions/contexts#runner-context